src/json/def_to_json.cr

```diff
@@ -0,0 +1,186 @@
+module JSON
+  # The `def_to_json` macro generates a `to_json(json : JSON::Builder)` method based on provided *mappings*.
+  #
+  # It is a lightweight alternative to `JSON.mapping` if you don't need to declare instance variables and a parser.
+  #
+  # The generated method invoks `to_json(JSON::Builder)` on each of the values returned by the *value* expression,
+  # or - if a converter is specified - `to_json(value, JSON::Builder)` on the converter.
+  #
+  # ### Example
+  #
+  # ```
+  # require "json"
+  #
+  # record Location, lat : Float64, long : Float64 do
+  #   JSON.def_to_json([lat, long])
```

Comment:
Where is this array stuff defined?

Reply:
Oh, I forgot to change this. The array syntax was not really useful.